gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/partitioned/RedundancyLogger.java

```diff
@@ -358,7 +360,16 @@ private void logWaitingForMembers() {
       /*
        * Log any offline members the region is waiting for.
        */
-      if(thereAreBucketsToBeRecovered && !offlineMembers.isEmpty()) {
+
+      if (firstNotification) {
+        if (sysCb != null) {
+          sysCb.waitingForDataSync(this.region, new HashSet(),
+              new HashSet(), this.thisMember, "");
+        }
+        this.firstNotification = false;
+      }
+      else if((thereAreBucketsToBeRecovered && !offlineMembers.isEmpty())) {
+
```

Comment:
Can you explain in brief why this condition "thereAreBucketsToBeRecovered && !offlineMembers.isEmpty()" is not being satisfied for colocated buckets? And shouldn't lack of notification result in a kind of hang rather than NPE?

Reply:
It may or maynot as all recovery is happening in parallel. Problem was we were getting extra notification for parent. Earlier parent's initialization was done without recovering buckets and now because it is waiting for buckets.
Lack of notification can cause hang. So made sure that, atleast first time all the regions are notified. Only after that, we check for bucketsTobeRecovered and offLineMembers etc.